ballista/rust/scheduler/src/scheduler_server/grpc.rs

```diff
@@ -390,7 +390,7 @@ impl<T: 'static + AsLogicalPlan, U: 'static + AsExecutionPlan> SchedulerGrpc
                     .and_then(|m| {
                         m.try_into_logical_plan(
                             session_ctx.deref(),
-                            self.codec.logical_extension_codec(),
+                            self.state.codec.logical_extension_codec(),
```

Comment:
Sorry, I'm daft - why does the codec move out of state? It needs to be unique to the scheduler instance for some reason?

Reply:
Since the codec is already a field of the state and is a necessary field for the state, I think it's better for the state to own the codec rather than the server. And it's better to keep unique instance of the codec in case that we change it in the future.

Reply:
Oh, I'm sorry, I got the direction wrong here: it went into state - a mistake with my diff tool :+1: